steepshot_io/graph/views.py

```diff
@@ -39,110 +55,72 @@ def group_steem_golos(self, list_steem, list_golos):
             idx += 1
         return res
 
-    def get_posts_count_monthly(self, platform=None):
-
+    def _get_data(self, platform=None, reverse=True, data_x=None, data_y=None, name_url=None, return_dict=False):
+        list_urls = []
         if platform == 'steem':
-            res = requests.get(settings.REQUESTS_URL_STEEM['posts_count_monthly']).json()
-            res.reverse()
-            values = []
-            for i in res:
-                values.append([i['date_to'], i['posts_count']])
-            return values
-        elif platform =='golos':
-            res = requests.get(settings.REQUESTS_URL_GOLOS['posts_count_monthly']).json()
-            res.reverse()
-            values = []
-            for i in res:
-                values.append([i['date_to'], i['posts_count']])
-            return values
+            list_urls.append(settings.STEEM_V1)
+        elif platform == 'golos':
+            list_urls.append(settings.GOLOS_V1)
         else:
-            res_steem = requests.get(settings.REQUESTS_URL_STEEM['posts_count_monthly']).json()
-            res_steem.reverse()
-            res_golos = requests.get(settings.REQUESTS_URL_GOLOS['posts_count_monthly']).json()
-            res_golos.reverse()
-            values_steem = []
-            values_golos = []
-            for i in res_steem:
-                values_steem.append([i['date_to'], i['posts_count']])
-            for i in res_golos:
-                values_golos.append([i['date_to'], i['posts_count']])
-            steem_golos_value = self.group_steem_golos(values_steem, values_golos)
-
-            return steem_golos_value
+            list_urls.extend([settings.STEEM_V1, settings.GOLOS_V1])
+
+        res = []
+        for url in list_urls:
+            try:
+                cur_url = settings.REQUESTS_URL[name_url].format(url=url)
+                get_data = requests.get(cur_url).json()
+                if return_dict:
+                    get_data = get_data['result']
+            except JSONDecodeError as e:
+                logger.error('Failed to parse json {}'.format(e))
+            except (ConnectionError, HTTPError) as e:
+                logger.error('Failed to connect to {}'.format(e))
+            except Exception as e:
+                logger.error(e)
+            if reverse:
+                get_data.reverse()
+            res.append([(i[data_x], i[data_y]) for i in get_data])
+        values_total = [i for i in res]
+        return values_total
 
     def get(self, request):
         if 'platform' in request.GET:
```

Comment:
If you will unify method `group_steem_golos` you can remove duplicate code and just write something like this:
``` python
    def get(self, request):
        values = self.group_steem_golos(
            *self._get_data(
                platform=request.GET.get('platform'),
                name_url='posts_count_monthly',
                data_x='date_to',
                data_y='posts_count'
            )
        )
        return render(request, self.template_name, {'values': values, 'name_1': 'counts posts'})
```